clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1247,7 +1240,10 @@ public DescribeClusterResult describeCluster(DescribeClusterOptions options) {
             void handleResponse(AbstractResponse abstractResponse) {
                 MetadataResponse response = (MetadataResponse) abstractResponse;
                 describeClusterFuture.complete(response.brokers());
-                controllerFuture.complete(response.controller());
+                Node controller = null;
+                if (response.controller() != null && response.controller().id() == MetadataResponse.NO_CONTROLLER_ID)
```

Comment:
Should be `response.controller().id() != MetadataResponse.NO_CONTROLLER_ID`, right?